core/commands/show_commit.py

```diff
@@ -8,18 +8,23 @@
 
 class GsShowCommitCommand(WindowCommand, GitCommand):
 
-    def run(self, commit_hash):
+    def run(self, commit_hash, show_file=False, filepath=None):
+        self.show_file = show_file
         repo_path = self.repo_path
         view = self.window.new_file()
         view.set_syntax_file("Packages/GitSavvy/syntax/show_commit.tmLanguage")
```

Comment:
Because of this line, the view will be syntax-highlighted as a commit (diff syntax plus a few other things) instead of the correct file-type.  For example, if you looked at the historical version of `path/to/my/file.py`, it would not be syntax-highlighted as a Python file.

Because of this, I think there is too little commonality between the original `GsShowCommitCommand` and what you're attempting to add.  Instead, create a new command that is completely distinct, `GsShowFileAtCommitCommand`.